### PR TITLE
Finished Info cog for api rewrite

### PR DESCRIPTION
**Pull request by smathep**
Re-added the info cog to the Clembot re-written architecture 
Syntax:

!info <user mention or ID>
!info

### CODE REVIEW

File: ClemBot.Bot/bot/cogs/info_cog.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+class InfoCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @ext.command()
+    @ext.required_claims(Claims.moderation_infraction_view)
+    @ext.ignore_claims_pre_invoke()
+    async def info(self, ctx, user: discord.User = None):
+        #Default range for message count 30 days(~1 month)
+        DEFAULT_MESSAGE_RANGE = 30
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/cogs/info_cog.py:20]: constants like this should be at the top under imports

File: ClemBot.Bot/bot/cogs/info_cog.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                moderation_info = f'» **Warnings:** {0 if warnings is None else len(warnings)} '
+                moderation_info += f'\n» **Mutes:** {0 if mutes is None else len(mutes)}'
+                moderation_info += f'\n» **Total Infractions:** {0 if infractions is None else len(infractions)}'
+                embed.add_field(name='**Moderation Info:**', value=moderation_info)
+
+            guild_info = ''
+            if (ctx.command.claims_check(claims)) or (ctx.author.id == user.id):
+                #prints the datetime in the format of <Months> <day> <year> <hours>:<minutes>:<seconds> <AM/PM>
+                guild_info = '» **Joined:** ' + member.joined_at.strftime('%b %d %Y %I:%M:%S %p') 
+            messages = (await self.bot.message_route.range_count_messages(user.id, ctx.guild.id, DEFAULT_MESSAGE_RANGE))['messageCount']
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/cogs/info_cog.py:61]: add new lines after ifs and loops to keep it from being one long block of code

File: ClemBot.Bot/bot/api/message_route.py [hunk 3]
```diff
@@ -62,3 +64,11 @@ async def batch_edit_message(self, messages: t.Iterable, **kwargs):
 
     async def get_message(self, message_id: int):
         return await self._client.get(f'messages/{message_id}')
+
+    async def range_count_messages(self, user_id: int, guild_id: int, days: int):
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/api/message_route.py:68]: this function should return an int, the consumer of it doesnt want a dict with a number, just the number

File: ClemBot.Bot/bot/cogs/info_cog.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            for i in member.roles[1:]:
+                guild_info += f'{i.mention}\t'
+            guild_info += f'\n» **Highest role:** {member.top_role.mention}'
+            guild_info += '\n» **Nitro boost date:** ' + (member.premium_since.strftime('%b %d %Y %I:%M:%S %p') if member.premium_since is not None else 'Not boosting')
+            embed.add_field(name='**Guild Information:**', value=guild_info, inline=False)
+
+        embed.set_footer(text=f'{ctx.author.name}#{ctx.author.discriminator}', icon_url=ctx.author.avatar_url)
+        msg = await ctx.send(embed=embed)
+        await self.bot.messenger.publish(Events.on_set_deletable, msg=msg, author=ctx.author)
+        return
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/cogs/info_cog.py:75]: we dont need this return

**Review by Jay-Madden — CHANGES_REQUESTED** [review]

File: ClemBot.Bot/bot/api/message_route.py [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import typing as t
 
+from discord import user
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/api/message_route.py:3]: Why do we need this?

Comment (smathep) [reply ClemBot.Bot/bot/api/message_route.py:3]: I think it was added by VSCode automatically sometime during my trials of how to do this lol

Comment (smathep) [reply ClemBot.Bot/bot/api/message_route.py]: deleted the line and tested. works fine

File: ClemBot.Bot/bot/api/message_route.py [hunk 6]
```diff
@@ -62,3 +61,11 @@ async def batch_edit_message(self, messages: t.Iterable, **kwargs):
...
     async def get_message(self, message_id: int):
         return await self._client.get(f'messages/{message_id}')
+
+    async def range_count_messages(self, user_id: int, guild_id: int, days: int):
+        json = {
+            'UserId' : user_id,
+            'GuildId' : guild_id,
+            'Days' : days
+        }
+        return (await self._client.get('messages/Count', data=json))['messageCount']
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/api/message_route.py:71]:
```suggestion
        resp = await self._client.get('messages/Count', data=json)
        
        if not resp:
        	return
        	
        return resp['messageCount']
```

Comment (smathep) [reply ClemBot.Bot/bot/api/message_route.py]: made change in following commit

**Review by Jay-Madden — APPROVED** [review]
lgtm

**Pull request merged**